3.10/arangosearch-nested-search.md

```diff
@@ -0,0 +1,221 @@
+  You can search for nested objects in arrays that satisfy multiple conditions
+  each, and define how often these conditions should be fulfilled for the entire
+  array
+title: Nested Search ArangoSearch Examples
+---
+# Nested search with ArangoSearch
+
+{{ page.description }}
+{:class="lead"}
+
+{% include hint-ee.md feature="Nested search with View index acceleration" %}
+
+By default, ArangoSearch Views index arrays as if the parent attribute had
+multiple values at once. With `trackListPositions` set to `true`, every array
+element is indexed individually and can be queried separately using the
+respective array index. With the nested search feature, you get another
+option for indexing arrays, in particular nested objects in arrays.
+
+You can let the View index the sub-objects in a way that lets you query for
+co-occurring values. For example, you can search the sub-objects and all the
+conditions need to be met by a single sub-object instead of across all of them.
+
+## Using nested search
+
+Consider the following document:
+
+```json
+{
+  "dimensions": [
+    { "type": "height", "value": 35 },
+    { "type": "width", "value": 60 }
+  ]
+}
+```
+
+You would normally index the `dimensions` field and its sub-fields with an
+`arangosearch` View definition like the following:
+
+```json
+{
+  "links": {
+    "<collection>": {
+      "fields": {
+        "dimensions": {
+          "fields": {
+            "type": {},
+            "value": {}
+          }
+        }
+      }
+    }
+  },
+  ...
+}
+```
+
+You might then write a query like the following to find documents where the
+height is greater than 40:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions.type == "height" AND doc.dimensions.value > 40
+  RETURN doc
+```
+
+This query matches the above document despite the height only being 35. The reason is
+that each condition is true for at least one of the nested objects. There is no
+check whether both conditions are true for the same object, however. You could
+add a `FILTER` statement to remove false positive matches from the search
+results, but it is cumbersome to check the conditions again, for every sub-object:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions.type == "height" AND doc.dimensions.value > 40
+  FILTER LENGTH(doc.dimensions[* FILTER CURRENT.type == "height" AND CURRENT.value > 40]) > 0
+  RETURN doc
+```
+
+The nested search feature allows you to condense the query while utilizing the
+View index:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions[? FILTER CURRENT.type == "height" AND CURRENT.value > 40]
+  RETURN doc
+```
+
+The required `arangosearch` View definition for this to work is as follows:
+
+```json
+{
+  "links": {
+    "<collection>": {
+      "fields": {
+        "dimensions": {
+          "nested": {
+            "type": {},
+            "value": {}
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Note the `nested` property instead of a `fields` property, configuring the View
```

Comment:
Is `nested` available in the Enterprise Edition only?

Reply:
To be precise, you can use the question mark operator in the community edition and it will give you the expected results, but it won't get any index acceleration, because that is an enterprise feature. I haven't checked whether the nested property is rejected/ignored in CE or if it's accepted and persisted, but it's definitely useless.

Reply:
Jörg to clarify how to word the docs regarding the different editions.